src/wp-includes/formatting.php

```diff
@@ -3161,7 +3155,42 @@ function wp_rel_callback( $matches, $rel ) {
 		}
 		$text = trim( $html );
 	}
-	return "<a $text rel=\"" . esc_attr( $rel ) . '">';
+
+	$rel_attr = $rel ? ' rel="' . esc_attr( $rel ) . '"' : '';
+
+	return "<a {$text}{$rel_attr}>";
+}
+
+/**
+ * Helper function used to determine if a URL is an internal URL.
+ *
+ * @since [version]
+ *
+ * @param string $href The URL to test.
+ * @return bool Returns true for internal URLs and false for all other URLs.
+ */
+function _wp_is_url_internal( $href ) {
+
+	if ( in_array( strtolower( wp_parse_url( $href, PHP_URL_SCHEME ) ), array( 'http', 'https' ), true ) ) {
```

Comment:
Is there any reason to not use `wp_allowed_protocols()` here? Not sure if URLs like `mailto:user@site.com` being noted as `nofollow ugc` is flagged as problematic.

Reply:
> Is there any reason to not use wp_allowed_protocols() here?
I didn't use it because I wasn't aware of it's existence. 

I'll update to use.

> Not sure if URLs like mailto:user@site.com being noted as nofollow ugc is flagged as problematic.

I'm not an expert here so I did a bit of googling and found a few thinks on various stack exchanges stating that in general crawlers don't crawl `mailto:` links so `nofollow` wouldn't be necessary *but* I don't think it would harm. But, semantically maybe that means that instead of using `wp_is_url_internal()` to determine the `rel` as a binary (either `ugc` or `ugc nofollow`) a helper function such as `wp_get_link_rel_attr()` (or something to that effect) should be created to handle this. This function could use the `wp_is_internal_link()` boolean return but also add additional conditions for links with `mailto:` protocol (and I'll look through other protocols to see if there might be any other conditions that should be handled differently).

Looks like a `mailto:` link should probably be `ugc` if in a comment but otherwise the nofollow would be superfluous. Maybe this is an unnecessary detail though.